plugins/pubIds/urn/URNPubIdPlugin.php

```diff
@@ -353,7 +354,7 @@ public function addPublicationFormFields(string $hookName, FormComponent $form):
                 'value' => $form->publication->getData('pub-id::other::urn'),
                 'prefix' => $prefix,
                 'pattern' => $pattern,
-                'contextInitials' => $form->submissionContext->getData('acronym', $form->submissionContext->getData('primaryLocale')) ?? '',
+                'contextInitials' => preg_replace('/[^-._;()\/A-Za-z0-9]/', '', Str::lower($form->submissionContext->getData('acronym', $form->submissionContext->getData('primaryLocale')) ?? '')),
```

Comment:
Do you know the purpose of this regular expression?
As it allows some special characters, it's a bit curious for me... Also it will remove a lot of valid letters (Á, Ç, Я, 火, etc).

Reply:
I don't know how the `contextInitials` will be used down the road by the front end, and without this context it's difficult to say what should be done... Without this context, I'd say it should just leave characters and digits or perhaps be removed.

Reply:
The issues where this was introduced (for DOIs): 
https://github.com/pkp/pkp-lib/issues/5505
https://github.com/pkp/pkp-lib/issues/7595

Reply:
@jonasraoni, OK so?

Reply:
Thanks, I'll approve the PR.

I still have lazy doubts (I could investigate by myself haha), which could be handled on another issue:
- Was the `URNPubIdPlugin` created only to generate DOIs for Crossref? If not, then this regular expression should be in another place.
- Is it ok if the regular expression leaves the `contextInitials` empty? If not, then we need to input something there.

Reply:
URN plugin is slightly different from DOIs and Crossref, but we treat the pattern defined by user for URN suffix in the same way as for DOIs. That means if the user defines that URN suffix should contain context initials (%j) it should behave in the same way as that (%j) in DOIs. URNs are however slightly different, e.g. they also consider if a check number needs to be added to the URN.
I suppose that the JMs are responsible to have the URN suffix patterns and the data needed for it set up properly. So if they for example use the default pattern that contains %j, and do not enter anything for contextInitials, then this is how they want their URNs to look like.